content/data/scripts/ui_topics.lua

```diff
@@ -21,24 +22,47 @@ local function weaponStats(weaponClass)
 	local baseDamage = weaponClass.Damage
 	if weaponClass.OuterRadius > 0 then
 		-- This weapon has a shockwave, which gives it additional damage on a direct hit.
-		-- TODO: Most weapons have shockwave damage equal to their base damage, but not all.
-		local bonusDamage = baseDamage
+		-- Added formula to calculate shockwave damage -- WW
+		if weaponClass.ShockwaveDamage and weaponClass.ShockwaveDamage > 0 then
+			bonusDamage = weaponClass.ShockwaveDamage
+		elseif weaponClass.ShockwaveDamage == 0 or weaponClass.ShockwaveDamage == nil then
```

Comment:
This can just be `else` instead of `elseif` which would also cover the unlikely chance that shockwave damage is < 0. Also is this correct? It translates to "If there's a shockwave with no damage then baseDamage = baseDamage * 2".

Reply:
That is an edge case I hadn't thought about. I will have to rewrite this check.

Reply:
I added the `else` instead of `elseif` but otherwise nothing can be done about this at this time because FSO cannot distinguish between a shockwave damage that is set to 0 and a shockwave damage that is not specified.

Reply:
Added the ba.print() warning for this scenario as you mentioned in the Discord. Marking this resolved for now.